apps/oc_chef_authz/src/oc_chef_policy.erl

```diff
@@ -0,0 +1,182 @@
+         record_fields/0,
+         set_created/2,
+         set_updated/2,
+         type_name/1,
+         update_from_ejson/2,
+         update_query/0,
+         update/2,
+         fetch/2,
+         route_name/1
+        ]).
+
+id(#oc_chef_policy{id = Id}) ->
+    Id.
+
+name(#oc_chef_policy{name = Name}) ->
+    Name.
+
+policy_group(#oc_chef_policy{policy_group = PolicyGroup}) ->
+    PolicyGroup.
+
+org_id(#oc_chef_policy{org_id = OrgId}) ->
+    OrgId.
+
+type_name(#oc_chef_policy{}) ->
+    policy.
+
+authz_id(#oc_chef_policy{authz_id = AuthzId}) ->
+    AuthzId.
+
+
+create_query() ->
+    insert_policy.
+
+update_query() ->
+    update_policy_by_name_group.
+
+delete_query() ->
+    delete_policy_by_name_group_org_id.
+
+find_query() ->
+    find_policy_by_orgid_name_group.
+
+list_query() ->
+    list_policies_for_org.
+
+bulk_get_query() ->
+    %% TODO: do we need this?
+    ok.
+
+new_record(OrgId, AuthzId, PolicyData) ->
+    Name = ej:get({<<"name">>}, PolicyData),
+    Id = ej:get({<<"id">>}, PolicyData, chef_object_base:make_org_prefix_id(OrgId, Name)),
+    PolicyGroup = ej:get({<<"policy_group">>}, PolicyData),
+    #oc_chef_policy{
+       id = Id,
+       authz_id = AuthzId,
+                       org_id = OrgId,
+                       name = Name,
+                       policy_group = PolicyGroup,
+      serialized_object = PolicyData}.
+
+create_record(OrgId, Name, RequestingActorId) ->
+    Policy = #oc_chef_policy{
+                           org_id = OrgId,
+                           name = Name},
+    set_created(Policy, RequestingActorId).
+
+set_created(#oc_chef_policy{} = Object, ActorId) ->
+    Object#oc_chef_policy{last_updated_by = ActorId}.
+
+set_updated(#oc_chef_policy{} = Object, ActorId) ->
+    Object#oc_chef_policy{last_updated_by = ActorId}.
+
+is_indexed() ->
+    false.
+
+ejson_for_indexing(#oc_chef_policy{}, _EjsonTerm) ->
+   {[]}.
+
+update_from_ejson(#oc_chef_policy{} = Policy, PolicyData) ->
+    Name = ej:get({<<"name">>}, PolicyData, name(Policy)),
+    Group = ej:get({<<"policy_group">>}, PolicyData, policy_group(Policy)),
+    Policy#oc_chef_policy{name = Name,
+                         policy_group = Group}.
+
+fields_for_update(#oc_chef_policy{
+                     last_updated_by = LastUpdatedBy,
+                     serialized_object = SerializedObject
+                                 } = Policy) ->
+    [name(Policy), policy_group(Policy), LastUpdatedBy, SerializedObject].
+
+
+fields_for_fetch(#oc_chef_policy{org_id = OrgId} = Policy) ->
+    [name(Policy), policy_group(Policy), OrgId].
+
+record_fields() ->
+    record_info(fields, oc_chef_policy).
+
+list(#oc_chef_policy{org_id = OrgId}, CallbackFun) ->
+    CallbackFun({list_query(), [OrgId], rows}).
+
+update(#oc_chef_policy{
+                      org_id = _OrgId,
+                      authz_id = _PolicyAuthzId,
+                      last_updated_by = _AuthzId
+                     } = _Record, _CallbackFun) ->
+    1.
```

Comment:
For now, you can `PUT $base_url/policies/:group_name/:policy_name` and that makes the data you sent live at that URL (regardless of create vs. update, etc.). In the next rev of the API (before we go GA with the feature overall) there's likely going to be a notion of "revision id" for a policy and it's probably going to be something immutable, like a SHA of the policyfile content.